pkg/pullsync/pullsync.go

```diff
@@ -475,26 +513,36 @@ func (s *Syncer) CancelRuid(ctx context.Context, peer swarm.Address, ruid uint32
 // handler handles an incoming request to explicitly cancel a ruid
 func (s *Syncer) cancelHandler(ctx context.Context, p p2p.Peer, stream p2p.Stream) (err error) {
 	r := protobuf.NewReader(stream)
+	var c pb.Cancel
 	defer func() {
 		if err != nil {
 			_ = stream.Reset()
+			if logMore {
+				s.logger.Debugf("pullsync: peer %s failed cancelling %d: %v", p.Address.String(), c.Ruid, err)
+			}
 		} else {
 			_ = stream.FullClose()
 		}
 	}()
 
-	var c pb.Cancel
 	if err := r.ReadMsgWithContext(ctx, &c); err != nil {
 		return fmt.Errorf("read cancel: %w", err)
 	}
 
+	if logMore {
+		s.logger.Debugf("pullsync: peer %s cancelling %d", p.Address.String(), c.Ruid)
+	}
+
 	s.ruidMtx.Lock()
 	defer s.ruidMtx.Unlock()
 
-	if cancel, ok := s.ruidCtx[c.Ruid]; ok {
+	if cancel, ok := s.ruidCtx[p.Address.ByteString()][c.Ruid]; ok {
```

Comment:
Still being a novice golang coder, does this handle if an unknown (s.ruidCtx[p.Address.ByteString()] is null) peer from asking us to cancel a [c.Ruid]?  Or will the if try to access nil[c.Ruid]?

Reply:
https://play.golang.org/p/hm2GrMxoTJA should be fine :+1:

Reply:
Just echoing years of my C coding where you always validate the first map/subscript is not null before referencing the next level down.